pallets/account/src/mock.rs

```diff
@@ -0,0 +1,239 @@
+}
+
+impl pallet_ethereum::Config for TestRuntime {
+    type RuntimeEvent = RuntimeEvent;
+    type StateRoot = pallet_ethereum::IntermediateStateRoot<Self>;
+    type PostLogContent = PostBlockAndTxnHashes;
+    type ExtraDataLength = ConstU32<30>;
+}
+
+parameter_types! {
+    pub TxWeightLimit: Weight = Weight::from_parts(u64::max_value(), 0);
+}
+
+impl pallet_account::Config for TestRuntime {
+    type RuntimeEvent = RuntimeEvent;
+    type Currency = Balances;
+    type DefaultAddressMapping = HashedAddressMapping<BlakeTwo256>;
+    type DefaultAccountMapping = HashedAccountMapping<BlakeTwo256>;
+    type ClaimSignature = EIP712Signature<Self, ChainId>;
+}
+
+pub(crate) type AccountId = AccountId32;
+pub(crate) type BlockNumber = u64;
+pub(crate) type Balance = u128;
+
+pub const ALICE: AccountId32 = AccountId32::new([0u8; 32]);
+pub const BOB: AccountId32 = AccountId32::new([1u8; 32]);
+pub const CHARLIE: AccountId32 = AccountId32::new([2u8; 32]);
+
+pub fn alice_secret() -> libsecp256k1::SecretKey {
+    libsecp256k1::SecretKey::parse(&keccak_256(b"Alice")).unwrap()
+}
+
+pub fn bob_secret() -> libsecp256k1::SecretKey {
+    libsecp256k1::SecretKey::parse(&keccak_256(b"Bob")).unwrap()
+}
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<TestRuntime>;
+type Block = frame_system::mocking::MockBlock<TestRuntime>;
+
+construct_runtime!(
+    pub struct TestRuntime
+    where
+        Block = Block,
+        NodeBlock = Block,
+        UncheckedExtrinsic = UncheckedExtrinsic,
+    {
+        System: frame_system,
+        Timestamp: pallet_timestamp,
+        Balances: pallet_balances,
+        Evm: pallet_evm,
+        Ethereum: pallet_ethereum,
+        Accounts: pallet_account,
+    }
+);
+
+pub struct ExtBuilder {
+    balances: Vec<(AccountId, Balance)>,
+}
+
+impl Default for ExtBuilder {
+    fn default() -> Self {
+        Self {
+            balances: vec![
+                (ALICE, 1_000_000_000_000),
+                (BOB, 1_000_000_000_000),
+                (CHARLIE, 1_000_000_000_000),
+            ],
+        }
+    }
+}
+
+impl ExtBuilder {
+    pub fn build(self) -> TestExternalities {
+        let mut t = frame_system::GenesisConfig::default()
+            .build_storage::<TestRuntime>()
+            .unwrap();
+
+        pallet_balances::GenesisConfig::<TestRuntime> {
+            balances: self.balances,
+        }
+        .assimilate_storage(&mut t)
+        .unwrap();
+
+        let mut ext = TestExternalities::from(t);
+        ext.execute_with(|| System::set_block_number(1));
+        ext
+    }
+
+    pub fn with_alice_mapping() -> TestExternalities {
+        let mut ext = Self::default().build();
+        ext.execute_with(|| {
+            let alice_eth = Accounts::eth_address(&alice_secret());
+            EvmAccounts::<TestRuntime>::set(ALICE, Some(alice_eth.clone()));
+            NativeAccounts::<TestRuntime>::set(alice_eth, Some(ALICE));
+        });
```

Comment:
You shouldn't do this since you're skipping the very functions you are supposed to be testing here.

Reply:
Still, I see this as bad practice.
You can ask for other opinions, of course.

If you have a public call to do something, you should use it instead of making "hacks".

Reply:
I understand, I'll change it to use public call.